python/triton/tools/aot.py

```diff
@@ -39,31 +39,34 @@
         print(module.str())
         sys.exit(0)
 
-    if not args.sm:
-        raise argparse.ArgumentError(None, "Must specify --sm for PTX compilation")
+    if 'ptx' in args.target:
+        if not args.sm:
+            raise argparse.ArgumentError(None, "Must specify --sm for PTX compilation")
 
-    # triton-ir -> triton-gpu-ir
-    module = triton.compiler.ttir_to_ttgir(module, num_warps=4, num_stages=3, compute_capability=args.sm)
-    if args.target == 'triton-gpu-ir':
-        print(module.str())
-        sys.exit(0)
+        # triton-ir -> triton-ptx-gpu-ir
+        module = triton.compiler.ttir_to_ttgir(module, num_warps=4, num_stages=3, compute_capability=args.sm)
+        if args.target == 'triton-ptx-gpu-ir':
+            print(module.str())
+            sys.exit(0)
 
-    # triton-gpu-ir -> llvm-ir
-    module = triton.compiler.ttgir_to_llir(module, extern_libs=None, compute_capability=args.sm)
-    if args.target == 'llvm-ir':
-        print(module)
-        sys.exit(0)
+        # triton-gpu-ir -> llvm-pts-ir
+        module = triton.compiler.ttgir_to_llir(module, extern_libs=None, compute_capability=args.sm)
+        if args.target == 'llvm-ptx-ir':
+            print(module)
+            sys.exit(0)
 
-    # llvm-ir -> ptx
-    if args.target == 'ptx':
-        if not args.ptx_version:
-            raise argparse.ArgumentError(None, "Must specify --ptx-version for PTX compilation")
-        module = triton.compiler.llir_to_ptx(module, compute_capability=args.sm, ptx_version=args.ptx_version)
+        # llvm-ir -> ptx
+        if args.target == 'ptx':
+            if not args.ptx_version:
+                raise argparse.ArgumentError(None, "Must specify --ptx-version for PTX compilation")
+            module = triton.compiler.llir_to_ptx(module, compute_capability=args.sm, ptx_version=args.ptx_version)
 
     # llvm-ir -> amdgcn
-    if args.target == 'amdgcn':
+    if 'amdgcn' in args.target:
         if not args.gfx:
             raise argparse.ArgumentError(None, "Must specify --gfx for AMDGCN compilation")
+        module = triton.compiler.ttir_to_ttgir(module, num_warps=4, num_stages=3, compute_capability=80)
```

Comment:
This is questionable.
I used 80, because this was the "default" compute capability, used in some other tests. for example in `Target/tritongpu_to_ptx.mlir`

Reply:
Removal of this hard coded compute capability for amd target requires refactoring, so I propose to do it as a separate task.